client/js/components/hyperdrive/client.js

```diff
@@ -0,0 +1,26 @@
+var path = require('path')
+var yofs = require('yo-fs')
+
+module.exports = function ui (root, entries, onclick) {
+  var lookup = {}
+  for (var i in entries) {
+    var entry = entries[i]
+    lookup[entry.name] = entry
+    var dir = path.dirname(entry.name)
+    if (!lookup[dir]) {
+      lookup[dir] = {
+        type: 'directory',
+        name: dir,
+        length: 0
+      }
+    }
+  }
+
+  var vals = []
+  for (var key in lookup) {
+    if (lookup.hasOwnProperty(key)) vals.push(lookup[key])
+  }
```

Comment:
The purpose of reading entries into an object and then into an array again is to deduplicate them, right?

The second part can be refactored to

```js
var vals = Object.keys(lookup).map(key => lookup[key])
```